packages/core/src/lib/metadata/index.ts

```diff
@@ -45,7 +41,9 @@ class Metadata extends BaseProtocol implements IMetadata {
    * Make a metadata query to a peer
    */
   public async query(peerId: PeerId): Promise<MetadataQueryResult> {
-    const request = proto_metadata.WakuMetadataRequest.encode(this.shardInfo);
+    const request = proto_metadata.WakuMetadataRequest.encode(
+      pubsubTopicsToShardInfo(this.pubsubTopics)
```

Comment:
nit: if `this.pubsubTopics` is not expected to change, then we can probably just create the shard info during construction and store it as another class variable